pkg/kubefed/init/init.go

```diff
@@ -321,6 +323,18 @@ func (i *initFederation) Run(cmdOut io.Writer, config util.AdminConfig) error {
 		return err
 	}
 
+	// Display a progress indicator, and exit if the overall init timeout is reached.
+	go func() {
+		err := wait.PollImmediate(2*time.Second, time.Duration(i.options.timeout)*time.Minute, func() (bool, error) {
+			fmt.Fprint(cmdOut, ".")
+			return false, nil
+		})
+		if err != nil {
```

Comment:
There are only one possible error here, which is timeout. What happens on that, only log message? We want the program to terminate when that happens, right?
Meanwhile @leahnp please check the changes and comments in [this PR](https://github.com/kubernetes/kubernetes/pull/42296) for reference.
Printing the status info is not that necessary.